lib/rouge/lexers/brightscript.rb

```diff
@@ -0,0 +1,140 @@
+          roBoolean roBoolean roBrightPackage roBrSub roButton roByteArray roCaptionRenderer 
+          roCaptionRendererEvent roCecInterface roCECStatusEvent roChannelStore 
+          roChannelStoreEvent roClockWidget roCodeRegistrationScreen 
+          roCodeRegistrationScreenEventroCompositor roControlDown roControlPort roControlPort 
+          roControlUp roCreateFile roDatagramReceiver roDatagramSender roDataGramSocket 
+          roDateTime roDeviceInfo roDeviceInfoEvent roDoubleroEVPCipher roEVPDigest 
+          roFileSystem roFileSystemEvent roFloat roFont roFontMetrics roFontRegistry roFunction 
+          roGlobal roGpio roGridScreen roGridScreenEvent roHdmiHotPlugEventroHdmiStatus 
+          roHdmiStatusEvent roHMAC roHttpAgent roImageCanvas roImageCanvasEvent roImageMetadata 
+          roImagePlayer roImageWidgetroInput roInputEvent roInt roInt roInvalid roInvalid 
+          roIRRemote roKeyboard roKeyboardPress roKeyboardScreen roKeyboardScreenEventroList 
+          roListScreen roListScreenEvent roLocalization roLongInteger roMessageDialog 
+          roMessageDialogEvent roMessagePort roMicrophone roMicrophoneEvent roNetworkConfiguration 
+          roOneLineDialog roOneLineDialogEventroParagraphScreen roParagraphScreenEvent roPath 
+          roPinEntryDialog roPinEntryDialogEvent roPinentryScreen roPosterScreen 
+          roPosterScreenEventroProgramGuide roQuadravoxButton roReadFile roRectangleroRegexroRegion 
+          roRegistry roRegistrySection roResourceManager roRSA roRssArticle roRssParser roScreen 
+          roSearchHistory roSearchScreen roSearchScreenEvent roSerialPort roSGNode roSGNodeEvent 
+          roSGScreenroSGScreenEvent roSlideShowroSlideShowEvent roSNS5 roSocketAddress roSocketEvent 
+          roSpringboardScreen roSpringboardScreenEventroSprite roStorageInfo roStreamSocket 
+          roStringroSystemLogroSystemLogEvent roSystemTime roTextFieldroTextScreen roTextScreenEvent 
+          roTextToSpeech roTextToSpeechEvent roTextureManager roTextureRequest 
+          roTextureRequestEventroTextWidget roTimer roTimespan roTouchScreen roTunerroTunerEvent 
+          roUniversalControlEvent roUrlEvent roUrlTransfer roVideoEvent roVideoInput roVideoMode 
+          roVideoPlayer roVideoPlayerEvent roVideoScreen roVideoScreenEventroWriteFile roXMLElement 
+          roXMLList 
+        )
+      end
+
+      id = /[$a-z_][a-z0-9_]*/io
+
+      state :root do
+        rule %r/\s+/m, Text::Whitespace
+
+        # https://developer.roku.com/en-ca/docs/references/brightscript/language/expressions-variables-types.md#comments
+        rule %r/(\'.*?$)/, Comment::Single
+        rule %r/((?i)REM.*?$)/, Comment::Single
+
+        # https://developer.roku.com/en-ca/docs/references/brightscript/language/expressions-variables-types.md#operators
+        rule %r([~!%^&*+=\|?:<>/-]), Operator
+
+        rule %r/\d+[lu]*/i, Num::Integer      
```

Comment:
This rule will match numbers occurring before a period and, given that `.` is lexed as `Punctuation`, this will mean that `3.14` will be tokenised as a combination of `Num::Integer` and `Punctuation` rather than `Num::Float`. This should come after the float regex to avoid this.

Reply:
Good catch, thanks. Updated.